src/backend/distributed/worker/worker_shard_visibility.c

```diff
@@ -372,6 +394,19 @@ ShouldHideShardsInternal(void)
 }
 
 
+/*
+ * IsPgBgWorker returns true if the current background worker
+ * belongs to Postgres.
+ */
+static bool
+IsPgBgWorker(void)
+{
+	Assert(MyBackendType == B_BG_WORKER);
+
+	return strcmp(MyBgworkerEntry->bgw_library_name, "postgres") == 0;
```

Comment:
this is documented / enforced here; https://github.com/postgres/postgres/blob/c6863b85829149e2241faafa161b6c5af1f06cb9/src/backend/postmaster/bgworker.c#L1232